ts/src/pro/hyperliquid.ts

```diff
@@ -161,6 +188,107 @@ export default class hyperliquid extends hyperliquidRest {
         return this.filterBySymbolSinceLimit (trades, symbol, since, limit, true);
     }
 
+    handleWsTickers (client: Client, message) {
+        //
+        //     {
+        //         "channel": "webData2",
+        //         "data": {
+        //             "meta": {
+        //                 "universe": [
+        //                     {
+        //                         "szDecimals": 5,
+        //                         "name": "BTC",
+        //                         "maxLeverage": 50,
+        //                         "onlyIsolated": false
+        //                     },
+        //                     ...
+        //                 ],
+        //             },
+        //             "assetCtxs": [
+        //                 {
+        //                     "funding": "0.00003005",
+        //                     "openInterest": "2311.50778",
+        //                     "prevDayPx": "63475.0",
+        //                     "dayNtlVlm": "468043329.64289033",
+        //                     "premium": "0.00094264",
+        //                     "oraclePx": "64712.0",
+        //                     "markPx": "64774.0",
+        //                     "midPx": "64773.5",
+        //                     "impactPxs": [
+        //                         "64773.0",
+        //                         "64774.0"
+        //                     ]
+        //                 },
+        //                 ...
+        //             ],
+        //             "spotAssetCtxs": [
+        //                 {
+        //                     "prevDayPx": "0.20937",
+        //                     "dayNtlVlm": "11188888.61984999",
+        //                     "markPx": "0.19722",
+        //                     "midPx": "0.197145",
+        //                     "circulatingSupply": "598760557.12072003",
+        //                     "coin": "PURR/USDC"
+        //                 },
+        //                 ...
+        //             ],
+        //         }
+        //     }
+        //
+        // spot
+        const data = this.safeDict (message, 'data', {});
+        const spotAssets = this.safeList (data, 'spotAssetCtxs', []);
+        const parsedTickers = [];
+        for (let i = 0; i < spotAssets.length; i++) {
+            const assetObject = spotAssets[i];
+            const ticker = this.parseWsTicker (assetObject, 'spot');
+            parsedTickers.push (ticker);
+        }
+        // perpetuals
+        const meta = this.safeDict (data, 'meta', {});
+        const universe = this.safeList (meta, 'universe', []);
+        const assets = this.safeList (data, 'assetCtxs', []);
+        for (let i = 0; i < assets.length; i++) {
+            let assetObject = assets[i];
+            const coinObject = universe[i];
+            const id = coinObject['name'] + ':USDC/USDC';
```

Comment:
@ttodua this is not correct, we're injecting as an exchange-specific field `coin` then `info` contains this field

```
'coin': 'kSHIB:USDC/USDC',
                              'dayNtlVlm': '2299289.598427',
                              'funding': '-0.00000479',
                              'impactPxs': ['0.019278', '0.019279']
```

Reply:
good catch, i missed that